_posts/pl/2021-12-14-http-idempotency.md

```diff
@@ -0,0 +1,41 @@
+---
+layout:    post
+title:     "Czy wiesz, czym jest idempotentność w metodach HTTP?"
+date:      2021-12-14 8:00:00 +0100
+published: true
+didyouknow: true
+lang: pl
+author:    fphilavong
+image:     /assets/img/posts/2021-12-14-http-idempotency/http-idempotency.jpg
+tags:
+- http
+---
+
+Idempotentność to właściwość pewnych operacji, która pozwala na ich wielokrotne stosowanie bez zmiany wyniku. Twórcy różnego rodzaju systemów rozproszonych bardzo chętnie korzystają z idempotentnych rozwiązań. Łatwiej jest w końcu powtórzyć wykonanie operacji, jeśli nie mamy pewności czy doszła do skutku, zamiast ją weryfikować.
+
+# Idempotentność w metodach HTTP
+Jeśli przy projektowaniu API stosujemy się do zasad `REST`, otrzymujemy idempotentność dla metod HTTP `GET`, `PUT`, `DELETE`, `HEAD`, `OPTIONS` i `TRACE`. Tylko interfejsy `POST` nie będą idempotentne.
+
+|Metoda HTTP|Idempotentność|
+|-----------|--------------|
+|OPTIONS    |tak           |
+|GET        |tak           |
+|HEAD       |tak           |
+|PUT        |tak           |
+|POST       |nie           |
+|DELETE     |tak           |
+|TRACE      |tak           |
+
+Przeanalizujmy, w jaki sposób powyższe metody HTTP stają się idempotentne i dlaczego `POST` nią nie jest.
+
+## POST
+Zazwyczaj (ale nie musi być to regułą) metoda `POST` służy do tworzenia nowego zasobu na serwerze. Więc kiedy wywołamy to samo żądanie `POST` *N* razy otrzymamy *N* nowych zasobów na serwerze. Tak więc `POST` nie jest idempotentny.
```

Comment:
_Metoda  `POST` służy zazwyczaj do tworzenia nowego zasobu na serwerze, chociaż nie jest to reguła. Kiedy wywołamy to samo żądanie `POST` *N* razy,  otrzymamy *N* nowych zasobów na serwerze. Tak więc `POST` nie jest idempotentny._